### PR TITLE
Feature/#5 text field component

### LINKED ISSUE #5: Implement Text Field Component
Implement [TextField](https://www.figma.com/design/M8nGARxOa4zTNUGPoDMHHo/Novix?node-id=234-6516&t=7MFBOhSlIYLVCbHh-4) in design system according to specified [details & measurements](https://www.figma.com/design/M8nGARxOa4zTNUGPoDMHHo/Novix?node-id=1521-1281&t=7MFBOhSlIYLVCbHh-4)

Image

Blocked by:
- #3

### PR DESCRIPTION
**Pull request by Hasa9nAli**
## Description
create NovixTextField in design system module that is essential component in the design system 

## Related Issue
#5 [Implement Text Field Component](https://github.com/Baghdad-Squad/Novix/issues/5)

## Screenshots 
https://github.com/user-attachments/assets/aa986f6b-f4dd-4a25-b58d-4191d64f24d1

## Feature
1. **Password Visibility Toggle**  
   - Set `isPassword = true`  
   - Verify that password is hidden by default (shows dots)  
   - Click the eye icon to toggle visibility  
   - Ensure text is revealed/hidden correctly  

2. **Leading Icon & Styling**  
   - Test with and without `leadingIcon`  
   - Verify icon tint changes on focus  

3. **Edge Cases**  
   - Test with empty input (placeholder visibility)  
   - Test `readOnly` and `enabled` states  
   - Test `maxLines` behavior

### CODE REVIEW

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import androidx.compose.ui.unit.dp
+import com.baghdad.design_system.R
+import com.baghdad.design_system.theme.Theme
+
+@Composable
+fun NovixTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    title: String = "",
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: label instead title

Comment (ahmednasserzaza) [reply design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: I see from the figma ui it's required everywhere, anyway, If it's optional make it nullable please

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    placeholder: String = "",
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: hint instead placeholder
and it should be nullable please and default value is null if it's already optional

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    @DrawableRes leadingIcon: Int? = null,
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: leadingIcon it should be better if it's a painter or a vectorResource

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    singleLine: Boolean = true,
+    isPassword: Boolean = false,
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: instead this, you can add isTextMasked = false
and you can handle visualTransformation like that

`                    visualTransformation = if (isTextMasked) PasswordVisualTransformation() else VisualTransformation.None,
`

Comment (ahmednasserzaza) [reply design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: in addition to, you have to pass trailingIcon and onClickTrailingIcon
boath should be nullable

**Review by abdulazizacc — COMMENTED** [review]
Well done Hasan, just do not merge the icons right now, wait it to be merged in the develop, then take it from there

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(Theme.color.surface, shape = RoundedCornerShape(12.dp))
+                    .border(
+                        1.dp,
+                        color = borderColor,
+                        shape = RoundedCornerShape(12.dp)
+                    )
+                    .clickable(
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt:102]: reuse clickable modifier @omer1998 created

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        targetValue = if (isFocused) Theme.color.primary else Theme.color.stroke,
+        animationSpec = tween(durationMillis = 400)
+    )
+
+    val iconTint by animateColorAsState(
+        targetValue = if (isFocused) Theme.color.primary else Theme.color.hint,
+        animationSpec = tween(durationMillis = 400)
+    )
+
+    var passwordVisible by remember { mutableStateOf(false) }
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: no need for this, it should be handled outside

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+                                )
+                            }
+
+                            Box(Modifier.weight(1f)) {
+                                if (value.isEmpty()) {
+                                    PlaceholderTextField(placeholder)
+                                }
+                                innerTextField()
+                            }
+                            if (isPassword) {
```
Comment (ahmednasserzaza) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: this should be handled from outside

File: design_system/src/main/res/drawable/eye.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="20dp"
+    android:height="20dp"
+    android:viewportWidth="20"
+    android:viewportHeight="20">
+  <path
+      android:pathData="M10,6.875C8.274,6.875 6.875,8.274 6.875,10C6.875,11.726 8.274,13.125 10,13.125C11.726,13.125 13.125,11.726 13.125,10C13.125,8.274 11.726,6.875 10,6.875ZM8.125,10C8.125,8.965 8.965,8.125 10,8.125C11.036,8.125 11.875,8.965 11.875,10C11.875,11.036 11.036,11.875 10,11.875C8.965,11.875 8.125,11.036 8.125,10Z"
+      android:fillColor="#ffffff"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/eye.xml]: make the fillColor =  '#0000'

File: design_system/src/main/res/drawable/eye.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+      android:fillAlpha="0.38"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/eye.xml]: remove fillAlpha

Comment (ahmednasserzaza) [reply design_system/src/main/res/drawable/eye.xml]: in addition to I think you need the opened and closed icons 
and please the naming should be 
ic_opened_eye or icon_opened_eye
ic_closed_eye or icon_closed_eye

File: design_system/src/main/res/drawable/eye.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+      android:fillType="evenOdd"/>
+  <path
+      android:pathData="M10,2.708C6.238,2.708 3.704,4.962 2.234,6.873L2.207,6.907C1.875,7.339 1.568,7.737 1.361,8.207C1.138,8.711 1.042,9.26 1.042,10C1.042,10.74 1.138,11.289 1.361,11.793C1.568,12.264 1.875,12.661 2.207,13.093L2.234,13.128C3.704,15.038 6.238,17.292 10,17.292C13.762,17.292 16.296,15.038 17.766,13.128L17.793,13.093C18.125,12.661 18.432,12.264 18.639,11.793C18.862,11.289 18.958,10.74 18.958,10C18.958,9.26 18.862,8.711 18.639,8.207C18.432,7.737 18.125,7.339 17.793,6.907L17.766,6.873C16.296,4.962 13.762,2.708 10,2.708ZM3.224,7.635C4.582,5.871 6.792,3.958 10,3.958C13.208,3.958 15.418,5.871 16.776,7.635C17.141,8.11 17.355,8.394 17.496,8.712C17.628,9.01 17.708,9.374 17.708,10C17.708,10.626 17.628,10.99 17.496,11.288C17.355,11.607 17.141,11.891 16.776,12.365C15.418,14.129 13.208,16.042 10,16.042C6.792,16.042 4.582,14.129 3.224,12.365C2.859,11.891 2.645,11.607 2.504,11.288C2.372,10.99 2.292,10.626 2.292,10C2.292,9.374 2.372,9.01 2.504,8.712C2.645,8.394 2.859,8.11 3.224,7.635Z"
+      android:fillColor="#ffffff"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/eye.xml]: same

File: design_system/src/main/res/drawable/eye.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+      android:fillAlpha="0.38"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/eye.xml]: same

File: design_system/src/main/res/drawable/ic_heroicons_outline.xml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+    android:width="20dp"
+    android:height="20dp"
+    android:viewportWidth="20"
+    android:viewportHeight="20">
+  <path
+      android:pathData="M3.317,6.852C2.547,7.762 1.96,8.83 1.612,10.001C2.688,13.615 6.036,16.25 9.999,16.25C10.826,16.25 11.627,16.135 12.385,15.921M5.19,5.19C6.57,4.28 8.223,3.75 10,3.75C13.963,3.75 17.311,6.385 18.387,9.999C17.794,11.993 16.509,13.69 14.81,14.81M5.19,5.19L2.5,2.5M5.19,5.19L8.232,8.232M14.81,14.81L17.5,17.5M14.81,14.81L11.768,11.768M11.768,11.768C12.22,11.315 12.5,10.69 12.5,10C12.5,8.619 11.381,7.5 10,7.5C9.31,7.5 8.685,7.78 8.232,8.232M11.768,11.768L8.232,8.232"
+      android:strokeAlpha="0.38"
+      android:strokeLineJoin="round"
+      android:strokeWidth="1.5"
+      android:fillColor="#00000000"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/ic_heroicons_outline.xml]: same

File: design_system/src/main/res/drawable/user.xml [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M12,1.25C9.1,1.25 6.75,3.601 6.75,6.5C6.75,9.399 9.1,11.75 12,11.75C14.899,11.75 17.25,9.399 17.25,6.5C17.25,3.601 14.899,1.25 12,1.25ZM8.25,6.5C8.25,4.429 9.929,2.75 12,2.75C14.071,2.75 15.75,4.429 15.75,6.5C15.75,8.571 14.071,10.25 12,10.25C9.929,10.25 8.25,8.571 8.25,6.5Z"
+      android:fillColor="#ffffff"
```
Comment (ahmednasserzaza) [inline design_system/src/main/res/drawable/user.xml]: same

**Review by ahmednasserzaza — CHANGES_REQUESTED** [review]
Well done, just check the comments please

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt (hunk 1 above)
Comment (Hasa9nAli) [reply design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: Screenshot 2025-07-03 at 8 41 26 PM

**some field not have lable**

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+    )
+
+
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp)
+    ) {
+        if (label != null) {
+            if (label.isNotEmpty()) {
```
Comment (MuhammedEdrees) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: These can be combined into single condition. Try to avoid nested if as much as possible as it make code harder to read.

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+    onClickTrailingIcon: (() -> Unit)? = null,
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    singleLine: Boolean = true,
+    isTextMasked: Boolean = false,
+    trailingVisibility: Boolean = false,
+    maxLines: Int = 1,
+) {
+    val interactionSource = remember { MutableInteractionSource() }
+    val isFocused by interactionSource.collectIsFocusedAsState()
```
Comment (MuhammedEdrees) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt:61]: Nice one👏. First time seeing this.

**Review by MuhammedEdrees — DISMISSED** [review]
Great work, Hasan 👏

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+        targetValue = if (isFocused) Theme.color.primary else Theme.color.stroke,
+        animationSpec = tween(durationMillis = 400)
+    )
+
+    val iconTint by animateColorAsState(
+        targetValue = if (isFocused) Theme.color.primary else Theme.color.hint,
+        animationSpec = tween(durationMillis = 400)
+    )
+
+
```
Comment (fatmahgazy) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: I think you need to delete extra space

File: design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+                                if (value.isEmpty()) {
+                                    hint?.let { HintTextField(it) }
+                                }
+                                innerTextField()
+                            }
+                            if (isTextMasked) {
+                                trailingIcon?.let {
+                                    Icon(
+                                        painter = it,
+                                        contentDescription = if (trailingVisibility) "Hide" else "Show",
```
Comment (fatmahgazy) [inline design_system/src/main/java/com/baghdad/design_system/component/NovixTextField.kt]: I think you can use string resources for text

**Review by fatmahgazy — DISMISSED** [review]
Great work Hassan 👏

**Review by MuhammedEdrees — APPROVED** [review]
Well done, Hasan👏

**Review by ahmednasserzaza — APPROVED** [review]

**Review by omer1998 — APPROVED** [review]
